src/VscPreviewExtension/src/childProcessHost.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { workspace, window, ExtensionContext, Uri } from "vscode";
+import * as childProcess from "child_process";
+import { ContentProvider } from "./contentProvider";
+
+export class ChildProcessHost {
+    public provider: ContentProvider;
+
+    protected _spawn: childProcess.ChildProcess;
+    protected _waiting: boolean;
+    private _content: string;
+    private _isMultipleRead = false;
+    protected _documentUri: Uri;
+    private ENDCODE = 7; // '\a'
+
+    constructor(context: ExtensionContext) {
+        // TODO: make path configurable
+        let extpath = context.asAbsolutePath("./DfmParse/Microsoft.DocAsCode.Dfm.VscPreview.exe");
+        this._spawn = childProcess.spawn(extpath);
+        if (!this._spawn.pid) {
+            window.showErrorMessage("Error:DfmProcess lost!");
+            return;
+        }
+        this._waiting = false;
+        this.initialProvider(context);
+        let that = this;
```

Comment:
for that in the closure stdout.on(){}(line 29~44) , `this` stand for another object( a Socket).

---
In reply to: [80382694](https://github.com/dotnet/docfx/pull/716#discussion_r80382694) [](ancestors = 80382694)